docs/api_reference/src/quantities.tex

```diff
@@ -1,266 +1,6669 @@
+                                           const @\exposconceptnc{QuantityConvertibleTo}<quantity>@ auto& rhs)
+  requires @\seebelownc@;
+friend constexpr decltype(auto) operator%=(@\exposconceptnc{Mutable}@<quantity> auto&& lhs,
+                                           const @\exposconceptnc{QuantityConvertibleTo}<quantity>@ auto& rhs)
+  requires @\seebelownc@;
+friend constexpr decltype(auto) operator*=(@\exposconceptnc{Mutable}@<quantity> auto&& lhs,
+                                           const @\exposconceptnc{ValuePreservingTo}@<Rep> auto& rhs)
+  requires @\seebelownc@;
+friend constexpr decltype(auto) operator*=(@\exposconceptnc{Mutable}@<quantity> auto&& lhs,
+                                           const @\libconcept{QuantityOf}@<dimensionless> auto& rhs)
+  requires @\seebelownc@;
+friend constexpr decltype(auto) operator/=(@\exposconceptnc{Mutable}@<quantity> auto&& lhs,
+                                           const @\exposconceptnc{ValuePreservingTo}@<Rep> auto& rhs)
+  requires @\seebelownc@;
+friend constexpr decltype(auto) operator/=(@\exposconceptnc{Mutable}@<quantity> auto&& lhs,
+                                           const @\libconcept{QuantityOf}@<dimensionless> auto& rhs)
+  requires @\seebelownc@;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let
+\begin{itemize}
+\item
+\tcode{@} be the \fakegrammarterm{operator},
+\item
+\tcode{\placeholdernc{TO-NUMERICAL-VALUE}(x)} be
+\tcode{x.\exposid{numerical-value}} for a quantity parameter and
+\tcode{x} otherwise,
+\item
+\tcode{common_rhs} be \tcode{rhs.in(lhs.unit)}
+if \tcode{@} is \tcode{+=}, \tcode{-=}, or \tcode{\%=},
+and \tcode{rhs} otherwise, and
+\item
+\tcode{RhsRep} be
+\tcode{std::remove_cvref_t<decltype(\placeholdernc{TO-NUMERICAL-VALUE}(rhs))>}.
+\end{itemize}
+
+\pnum
+\expects
+If \tcode{@} is \tcode{\%=} or \tcode{/=}, then
+\begin{codeblock}
+@\placeholdernc{TO-NUMERICAL-VALUE}@(rhs) != representation_values<RhsRep>::zero()
+\end{codeblock}
+is \tcode{true}.
+
+\pnum
+\effects
+Equivalent to
+\tcode{lhs.\exposid{numerical-value} @ \placeholdernc{TO-NUMERICAL-VALUE}(common_rhs)}.
+
+\pnum
+\returns
+\tcode{std::forward<decltype(lhs)>(lhs)}.
+
+\pnum
+\remarks
+Let \placeholder{C} be
+\begin{itemize}
+\item
+\tcode{(!treat_as_floating_point<rep>)} if \tcode{@} is \tcode{\%=},
+\item
+\tcode{(!\libconcept{Quantity}<std::remove_cvref_t<decltype(rhs)>>)}
+if \tcode{rhs} is not a quantity parameter,
+\item
+\tcode{(rhs.unit == mp_units::one)} if \tcode{rhs} is a \tcode{\libconcept{QuantityOf}<dimensionless>} parameter, and
+\item
+\tcode{true} otherwise.
+\end{itemize}
+The expression in the \fakegrammarterm{requires-clause} is equivalent to:
+\begin{codeblock}
+(@\placeholder{C}@) && @\exposconceptnc{ValuePreservingTo}@<RhsRep, Rep> && requires {
+  { lhs.@\exposid{numerical-value}@ @\atsign@ @\placeholdernc{TO-NUMERICAL-VALUE}@(rhs) } -> std::@\stdconcept{same_as}@<rep&>;
+}
```

Comment:
I don't require compound assignments in new representation concepts. Also, I do not check anything against another type there. So I need to provide the "Otherwise" case 😉

Reply:
> I don't require compound assignments in new representation concepts.

Oh, you're right.
That'd only work for unary `+`,
and maybe only until you support tensors again.

> Also, I do not check anything against another type there. So I need to provide the "Otherwise" case 😉

What do you mean?

Reply:
In the constraints you mentioned above, we checked if we can add `rep{} += Rep{}`. `Representation` concepts check for `rep + rep` (on the same type). This is why we need to repeat the checks for nearly every overload of `quantity`; we may typically deal with two different representation types there.